src/firejail/main.c

```diff
@@ -1498,8 +1501,11 @@ int main(int argc, char **argv, char **envp) {
 			arg_rlimit_sigpending = 1;
 		}
 		else if (strncmp(argv[i], "--rlimit-as=", 12) == 0) {
-			check_unsigned(argv[i] + 12, "Error: invalid rlimit");
-			sscanf(argv[i] + 12, "%llu", &cfg.rlimit_as);
+			cfg.rlimit_as = parse_arg_size(argv[i] + 12);
+			if ( cfg.rlimit_as == 0 ) {
+				perror("Error: given rlimit-as is invalid. use only non-negative numbers and k,m,g suffix for size");
```

Comment:
suggestion:
 perror("Error: invalid rlimit-as. Only use positive numbers and k,m or g suffix.");